### PR TITLE
Added CI/docs to Github actions

### PR DESCRIPTION
**Pull request by theogf**
This is a starting point to reproduce the current behavior of Travis

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+          - ubuntu-latest
+          - macOS-latest
+          - windows-latest
+        arch:
+          - x64
+        nightly: [false]
+        include:
+          - os: ubuntu-latest
+            version: 'nightly'
+            nightly: true
```
Comment (devmotion) [inline .github/workflows/ci.yml:26]: This seems a bit redundant, doesn't it? If you want to figure out if something runs with Julia nightly you could just check `matrix.version`:
```suggestion
        include:
          - os: ubuntu-latest
            version: 'nightly'
```

Comment (theogf) [reply .github/workflows/ci.yml:26]: Ah you mean I could write `continue-on-error: ${{ matrix.version == 'nightly' }}` ?

Comment (devmotion) [reply .github/workflows/ci.yml:26]: Yes

Comment (devmotion) [reply .github/workflows/ci.yml:26]: That's what we do in other Turing packages and it works :smiley:

Comment (theogf) [reply .github/workflows/ci.yml:26]: Nice!

File: README.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # KernelFunctions.jl
 
-[![Build Status](https://travis-ci.com/JuliaGaussianProcesses/KernelFunctions.jl.svg?branch=master)](https://travis-ci.com/JuliaGaussianProcesses/KernelFunctions.jl)
+![CI](https://github.com/JuliaGaussianProcesses/KernelFunctions.jl/workflows/CI/badge.svg)
```
Comment (devmotion) [inline README.md:3]:
```suggestion
![CI](https://github.com/JuliaGaussianProcesses/KernelFunctions.jl/workflows/CI/badge.svg?branch=master)
```

**Comment by theogf** [comment]
@devmotion Do you know why it's triggered by both push and pull_request?

**Comment by devmotion** [comment]
You pushed to a branch in the repo and opened a pull request, so that's expected :smile: Probably you only want to run CI tests if something is pushed to master or a pull request is opened. I added a suggestion that achieves this.

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
+name: CI
+on:
+  - push
```
Comment (devmotion) [inline .github/workflows/ci.yml:3]:
```suggestion
  - push
    branches:
      - master
```

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  - pull_request
```
Comment (devmotion) [inline .github/workflows/ci.yml:4]:
```suggestion
  push:
    branches:
      - master
  pull_request:
```

**Review by devmotion — APPROVED** [review]
LGTM

**Comment by devmotion** [comment]
Ah one thing I missed: I guess it would be good to delete `.travis.yml`?

**Comment by devmotion** [comment]
Ah wait it's actually deleted 😂

**Pull request merged**